### PR TITLE
Update homematic.markdown config variables style

### PR DESCRIPTION
**Pull request by rdbahm**
Per #6385 updating the configuration description.

**Description:**
I've needed to make a few guesses based on the documentation - Homematic's configuration examples are divided into three parts because it uses heavily nested configuration. It is not immediately obvious to me if the system can handle having multiple {% configuration %} blocks in a single page - let me know and I can always adjust how it's set up.

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Comment by frenck** [comment]
Hi there @rdbahm,

It seems like the build is failing, can you please take a look? 👍 

../Frenck

File: source/_components/homematic.markdown [hunk 1]
```diff
@@ -34,28 +34,90 @@ homematic:
...
-- **username** (*Optional*): When fetching names via JSON-RPC, you need to specify a user with guest-access to the CCU.
-- **password** (*Optional*): When fetching names via JSON-RPC, you need to specify the password of the user you have configured above.
-- **path** (*Optional*): Set to `/groups` when using port 9292.
+{% configuration %}
+host:
+  description: IP address or Hostname of CCU/Homegear device or Hass.io add-on.
+  required: true
+  type: string
+port:
+  description: Port of CCU/Homegear XML-RPC Server. Wireless: 2001, wired: 2000, IP: 2010, thermostatgroups: 9292.
```
Comment (WofWca) [inline source/_components/homematic.markdown]: There's a `:`. Quotation needed.

File: source/_components/homematic.markdown [hunk 2]
```diff
@@ -34,28 +34,90 @@ homematic:
...
+  required: false
+  type: string
+callback_port:
+  description: Set this, if Home Assistant is reachable under a different IP from the CCU (NAT, Docker etc.).
+  required: false
+  type: int
+resolvenames:
+  description: Try to fetch device names. Defaults to `false` if not specified.
+  required: false
+  type: map
```
Comment (WofWca) [inline source/_components/homematic.markdown]: I think it is incorrect to say that the type is `map` here. It is string, but with only a few valid values.

**Review by WofWca — CHANGES_REQUESTED** [review]
The conversion is incorrect.

File: source/_components/homematic.markdown [hunk 3]
```diff
@@ -34,28 +34,90 @@ homematic:
...
+  required: false
+  type: list
+local_ip:
+  description: IP of device running Home Assistant. Override auto-detected value for exotic network setups.
+  required: false
+  type: string
+local_port:
+  description: Port for connection with Home Assistant. By default it is randomly assigned.
+  required: false
+  type: int
```
Comment (frenck) [inline source/_components/homematic.markdown]: `integer`

File: source/_components/homematic.markdown (hunk 2 above)
Comment (frenck) [reply source/_components/homematic.markdown]: Indeed. @rdbahm this should be a `string`.

File: source/_components/homematic.markdown [hunk 4]
```diff
@@ -34,28 +34,90 @@ homematic:
...
+callback_ip:
+  description: Set this, if Home Assistant is reachable under a different IP from the CCU (NAT, Docker etc.).
+  required: false
+  type: string
+callback_port:
+  description: Set this, if Home Assistant is reachable under a different IP from the CCU (NAT, Docker etc.).
+  required: false
+  type: int
```
Comment (frenck) [inline source/_components/homematic.markdown]: `integer`

File: source/_components/homematic.markdown [hunk 5]
```diff
@@ -34,28 +34,90 @@ homematic:
...
-- **path** (*Optional*): Set to `/groups` when using port 9292.
+{% configuration %}
+host:
+  description: IP address or Hostname of CCU/Homegear device or Hass.io add-on.
+  required: true
+  type: string
+port:
+  description: Port of CCU/Homegear XML-RPC Server. Wireless: 2001, wired: 2000, IP: 2010, thermostatgroups: 9292.
+  required: false
+  type: int
```
Comment (frenck) [inline source/_components/homematic.markdown]: `integer`

File: source/_components/homematic.markdown [hunk 6]
```diff
@@ -34,28 +34,90 @@ homematic:
...
+callback_port:
+  description: Set this, if Home Assistant is reachable under a different IP from the CCU (NAT, Docker etc.).
+  required: false
+  type: int
+resolvenames:
+  description: Try to fetch device names. Defaults to `false` if not specified.
+  required: false
+  type: map
+  default: false
+  keys:
```
Comment (frenck) [inline source/_components/homematic.markdown]: I suggest restoring the part where those where explained (since this is not a map)

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_components/homematic.markdown (hunk 2 above)
Comment (rdbahm) [reply source/_components/homematic.markdown]: Makes sense. Is there a better way than "keys" to specify what the options are? (Does "keys"work without it being set to "map" for type?")

File: source/_components/homematic.markdown (hunk 3 above)
Comment (rdbahm) [reply source/_components/homematic.markdown]: Documentation says "int" for the type - is it incorrect? https://developers.home-assistant.io/docs/en/documentation_create_page.html

File: source/_components/homematic.markdown (hunk 2 above)
Comment (rdbahm) [reply source/_components/homematic.markdown]: Found the comment below where you suggest restoring the section - agreed, and I'll do that in the next commit.

**Comment by rdbahm** [comment]
Okay, I've made the requested changes. An outstanding question I had left (outside the scope of this one request) is "int" vs "integer." The documentation says to use "int" (https://developers.home-assistant.io/docs/en/documentation_create_page.html#configuration) while it looks like "integer" is what you're looking for. Should the documentation be updated?

**Comment by WofWca** [comment]
[It is updated](https://github.com/home-assistant/developers.home-assistant/blob/master/docs/documentation_create_page.md),  but the updates are not live yet.

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks for updating the PR @rdbahm! 👍 

Happy Hacktoberfest 🎉 

../Frenck

**Pull request merged**